10up/insecure-content-warning
# Highlight insecure element and scroll to it

### Description of the Change

Adds "View element" button, which scrolls editor to the element with insecure content. The element is highlighted with the red outline.

Works both in Classic and Block editor.

The outline is cleared before every new content check, post save or preview.

Closes #9

### Verification Process

1. Create new post
2. Add some content and put insecure element/block at the end of the content (outside of the first screen)
3. Try to save the post, the insecure warning should appear
4. Click on "View element" link/button (depends on the Block or Classic editor you use)
5. The page expected to scroll to the element/block and add a red outline to it

https://user-images.githubusercontent.com/288381/170232262-0026ef52-7567-45cd-aae0-494d78b81250.mp4

### Possible Drawbacks

The original tracking design doesn't allow to identify exact element/block which refers to insecure content. We only have list of URLs, which is used in this PR to find elements in the editor.

If the content have multiple elements with the same URL, multiple items with this URL will be added to the warning block. But any "View element" link from the identical URLs will only scroll/highlight the first element.

### Checklist:






### Changelog Entry

> Added - "View element" link to highlight and auto-scroll to the insecure element
> Fixed - Dependencies of JavaScript assets



### Credits


Props @cadic, @psorensen, @adamsilverstein, @dkotter
